server/chat-plugins/seasons.ts

```diff
@@ -159,22 +171,44 @@ export async function updateBadgeholders() {
 	if (!data.badgeholders[period]) {
 		data.badgeholders[period] = {};
 	}
-	for (const formatName of data.formatSchedule[findPeriod()]) {
+
+	const currentPeriod = findPeriod();
+	const scheduledFormats = data.formatSchedule[currentPeriod];
+	if (!scheduledFormats) {
+		Monitor.crashlog(new Error(`No format schedule found for period ${currentPeriod}`), "Season badge update");
+		return;
+	}
+
+	for (const formatName of scheduledFormats) {
 		const formatid = `gen${Dex.gen}${formatName}`;
 		const response = await getLadderTop(formatid);
-		if (!response) continue; // ??
+
+		// Ensure response is an array
+		if (!response || !Array.isArray(response)) {
+			continue;
+		}
```

Comment:
```suggestion
```
This endpoint will always serve an array. If for some reason it isn't, it should be erroring out loudly. You can probably just remove this check altogether, there shouldn't ever be a need for it.

Reply:
You're right; the check is unnecessary. I've removed it. Thanks for pointing that out.